src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -304,6 +304,10 @@ public async Task<bool> IsAllProjectsNominatedAsync()
             return true;
         }
 
+        public IReadOnlyList<object> GetAllProjectRestoreInfoSources()
```

Comment:
Should we convert getters and setters to properties ? In this case `GetAllProjectRestoreInfoSources()` to a property  `ProjectRestoreInfoResources`

Reply:
i didn't find how it was used, and why it is weak typed...

Reply:
Currently, this is consistent with the rest of the methods in this class.

Reply:
@lifengl All the main code is used in SolutionRestoreWorker. 

It's weakly typed to avoid a heavy refactoring of the cache.